packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -566,14 +567,14 @@ function getSlideToSegment(
 	segment: ISegment | undefined,
 	slidingPreference: SlidingPreference = SlidingPreference.FORWARD,
 	cache?: Map<ISegment, { seg?: ISegment }>,
-): ISegment | undefined {
-	if (!segment || !isRemovedAndAcked(segment)) {
-		return segment;
+): [ISegment | undefined, "start" | "end" | undefined] {
```

Comment:
I quite like this idea and did start implementing it -- but it ends up not working perfectly because you do need to have both values. A reference sliding to an endpoint doesn't mean that it would necessarily be detached, and so `ISegment | undefined` cannot be made equivalent to `ISegment | "start" | "end"`.

Reply:
For example, if we start with a string like `-a(bcd)e-`, with hyphens representing the endpoints, when we delete `abc`, the start reference will slide to an endpoint: `(-d)e-`. The endpoint-aware sliding doesn't look to the opposite direction when it runs out of segments in the given direction, unlike regular sliding.

Reply:
thanks for reminding me of that case. I remember thinking that it was odd that we tried to slide in both directions (despite writing that code a while back :P).

Do we actually care about the "try both directions" behavior for other use cases? I agree that we don't want it for intervals. I feel like we'd be fine with a single-direction, accumulate on end/start of string where appropriate approach for undo-redo as well. Maybe there are details to work out, but I think we can at least file some work to simplify this and make a note here that the model of how sliding works seems like it can be simplified a bit given this change. I do like not bundling that into this PR.

Reply:
I created https://dev.azure.com/fluidframework/internal/_workitems/edit/5205 to track simplifying the sliding logic and assigned it to myself as a placeholder.